sonic_psu/psu_base.py

```diff
@@ -47,3 +47,75 @@ def get_psu_presence(self, index):
         """
         return False
 
+    def get_model(self, idx):
+        """
+        Retrieves the model number/name of a power supply unit (PSU) defined
+                by 1-based index <idx>
+        :param idx: An integer, 1-based index of the PSU of which to query model number
+        :return: String, denoting model number/name
+        """
+        return ""
+
+    def get_mfr_id(self, idx):
+        """
+        Retrieves the manufacturing id of a power supply unit (PSU) defined
+                by 1-based index <idx>
+        :param idx: An integer, 1-based index of the PSU of which to query mfr id
+        :return: String, denoting manufacturing id
+        """
+        return ""
+
+    def get_serial(self, idx):
+        """
+        Retrieves the serial number of a power supply unit (PSU) defined
+                by 1-based index <idx>
+        :param idx: An integer, 1-based index of the PSU of which to query serial number
+        :return: String, denoting serial number of the PSU unit
+        """
+        return ""
+
+    def get_direction(self, idx):
+        """
+        Retrieves the airflow direction of a power supply unit (PSU) defined
+                by 1-based index <idx>
+        :param idx: An integer, 1-based index of the PSU of which to query airflow direction
+        :return: String, denoting the airflow direction
+        """
+        return ""
+
+    def get_output_voltage(self, idx):
+        """
+        Retrieves the ouput volatage in milli volts of a power supply unit (PSU) defined
+                by 1-based index <idx>
+        :param idx: An integer, 1-based index of the PSU of which to query o/p volatge
+        :return: An integer, value of o/p voltage in mV if PSU is good, else zero
+        """
+        return 0
+
+    def get_output_current(self, idx):
+        """
+        Retrieves the output current in milli amperes of a power supply unit (PSU) defined
+                by 1-based index <idx>
+        :param idx: An integer, 1-based index of the PSU of which to query o/p current
+        :return: An integer, value of o/p current in mA if PSU is good, else zero
+        """
+        return 0
+
+    def get_output_power(self, idx):
+        """
+        Retrieves the output power in micro watts of a power supply unit (PSU) defined
+                by 1-based index <idx>
+        :param idx: An integer, 1-based index of the PSU of which to query o/p power
+        :return: An integer, value of o/p power in micro Watts if PSU is good, else zero
+        """
+        return 0
+
+    def get_fan_rpm(self, idx, fan_idx):
```

Comment:
"get_fan_speed" ?

Reply:
Since fan speed unit is RPM, this name is chosen for the API. I can change this to 'get_fan_speed' if you think that represents the objective better.

Reply:
if you look at the function names in fan_base, they are using 'speed', I would suggest using 'speed' for consistency.

Reply:
Sure. I will update the PR